UI/EncyclopediaDetailPanel.cpp

```diff
@@ -140,6 +140,25 @@ namespace {
         return dir_names;
     }
 
+    /** Find Encyclopedia article with given name
+     * @param[in] name name entry of the article
+     */
+    EncyclopediaArticle GetPediaArticle(const std::string& name) {
+        const std::map<std::string, std::vector<EncyclopediaArticle> >& articles = GetEncyclopedia().articles;
+        for (std::map<std::string, std::vector<EncyclopediaArticle> >::const_iterator category_it = articles.begin();
+             category_it != articles.end(); ++category_it)
+        {
+            for (std::vector<EncyclopediaArticle>::const_iterator article_it = category_it->second.begin();
+                 article_it != category_it->second.end(); ++article_it)
+            {
+                if (article_it->name == name) {
+                    return *article_it;
+                }
+            }
+        }
+        return EncyclopediaArticle("", "", "", "", "");
```

Comment:
probably fine as is, but my initial reaction is that there should be a default constructor for this case...